syntax = "proto3";

//声明grpc服务
service Greeter {
   /*
   以下 分别是 服务端 推送流， 客户端 推送流 ，双向流。
   */

  // 通过在响应类型前插入 stream 关键字，可以指定一个服务端的流方法
  rpc GetStream (StreamReqData) returns (stream StreamResData){}


  // 通过在请求类型前插入 stream 关键字，可以指定一个客户端的流方法
  rpc PutStream (stream StreamReqData) returns (StreamResData){}
  rpc AllStream (stream StreamReqData) returns (stream StreamResData){}
}

//stream请求结构
message StreamReqData {
   string req_data = 1;
}
//stream返回结构
message StreamResData {
   string res_data = 1;
}



